### PR TITLE
Remove browser comparison pages [no bug]

### PR DESCRIPTION
**Pull request by craigcook**
## One-line summary
Taking down outdated pages until we can replace them with evergreen content.

## Testing
All pages should redirect to /firefox/new/

http://localhost:8000/firefox/browsers/compare/
http://localhost:8000/firefox/browsers/compare/brave/
http://localhost:8000/firefox/browsers/compare/chrome/
http://localhost:8000/firefox/browsers/compare/edge/
http://localhost:8000/firefox/browsers/compare/opera/
http://localhost:8000/firefox/browsers/compare/safari/

Check for any links to these pages I might have missed, or any references to deleted assets on other pages.

### CODE REVIEW

**Review by robhudson — APPROVED** [review]
🕸️

File: bedrock/firefox/templates/firefox/browsers/mobile/index.html [hunk 1]
```diff
@@ -142,19 +142,6 @@ <h3 class="mzp-c-menu-list-title">{{ ftl('browsers-mobile-download') }}</h3>
...
-    {% call split(
-      image=resp_img('img/firefox/browsers/dev.png',
-        srcset={ 'img/firefox/browsers/dev-high-res.png': '2x' },
-        optional_attributes={ 'class': 'mzp-c-split-media-asset'}
-      ),
-      block_class='t-compare'
-    ) %}
-      <h2>{{ ftl('browsers-mobile-see-how-firefox-for-desktop-compare') }}</h2>
-      <p class="c-compare-title mzp-has-zap-06">{{ ftl('browsers-mobile-see-how-firefox-for-desktop-strong') }}</p>
-      <p><a class="mzp-c-cta-link" href="{{ url('firefox.browsers.compare.index') }}" data-cta-type="link" data-cta-text="Compare">{{ ftl('browsers-mobile-compare') }}</a></p>
```
Comment (janbrasna) [inline bedrock/firefox/templates/firefox/browsers/mobile/index.html:155]: Q: should the fluent strings stay defined for now?
https://github.com/mozilla/bedrock/blob/1519e9d9b3824b53461fde3baa2ca2677df8f746/l10n/en/firefox/browsers/mobile/index.ftl#L25-L29

Comment (craigcook) [reply bedrock/firefox/templates/firefox/browsers/mobile/index.html:155]: The comparison pages are being rewritten and it's possible we'll be able to reuse some of these strings. We can probably leave them defined-but-unused for now until we've figured out the next steps.

Comment (janbrasna) [reply bedrock/firefox/templates/firefox/browsers/mobile/index.html:155]: That's what I thought. Only double checking;) 🥂

**Pull request merged**